db/hints/manager.cc

```diff
@@ -725,15 +714,26 @@ future<> manager::end_point_hints_manager::sender::send_one_hint(lw_shared_ptr<s
             } catch (no_column_mapping& e) {
                 manager_logger.debug("send_hints(): {} at {}: {}", fname, rp, e.what());
                 ++this->shard_stats().discarded;
+            } catch (...) {
+                manager_logger.debug("send_hints(): unexpected error in file {} at {}: {}", fname, rp, std::current_exception());
+                ++this->shard_stats().discarded;
```

Comment:
Please, don't discard this hint. 
The v2 version was better: 
 1) Call `ctx_ptr->on_hint_send_failure(rp);`
 2) Add a debug message.

And then simply add `return make_exception_future<>(std::current_exception());`